src/tools/compiletest/src/runtest.rs

```diff
@@ -2329,13 +2337,17 @@ impl<'test> TestCx<'test> {
         // useful flag.
         //
         // For now, though…
-        if let Some(rev) = self.revision {
-            let prefixes = format!("CHECK,{}", rev);
-            if self.config.llvm_version.unwrap_or(0) >= 130000 {
-                filecheck.args(&["--allow-unused-prefixes", "--check-prefixes", &prefixes]);
-            } else {
-                filecheck.args(&["--check-prefixes", &prefixes]);
-            }
+        let prefix_for_target =
+            if self.config.target.contains("msvc") { "MSVC" } else { "NONMSVC" };
+        let prefixes = if let Some(rev) = self.revision {
+            format!("CHECK,{},{}", prefix_for_target, rev)
+        } else {
+            format!("CHECK,{}", prefix_for_target)
+        };
```

Comment:
I think this change might not be needed as something similar already seems to be supported via "revisions". See https://github.com/rust-lang/rust/blob/b6f3cb9502b1910f6af32f426fdb78e813b390ef/src/test/codegen/fewer-names.rs for an example and https://rustc-dev-guide.rust-lang.org/tests/adding.html?#revisions for a description of revisions.

Reply:
I tried to use revisions but I don't think that they work here: as I understand it, revisions allow you to run the test multiple times with different compiler options/configurations, so I could have the test built twice (once for MSVC and once for GNU) but that requires that the compiler has been built with both the MSVC and GNU targets which may not be true on Windows and definitely won't work on Linux.

Looking at other tests, they seem to fork the same file into a "standard" variant and an MSVC variant and then use `// ignore-msvc` and `// only-msvc` - this seemed fragile (as there are now two test files to maintain) and difficult to understand the difference between the two (one has to realize that there are two files and then diff them (assuming they are kept in sync, otherwise the diff is a mess to parse)).

In future we may want a way to generalize this per target/backend (LLVM vs MSVC vs Cranelift vs GCC), but for now the MSVC vs non-MSVC at least avoids forking this file, and would allow us to unfork the other files.

Reply:
OK, yes avoiding the need to have separate test files seems like a good argument to support something like this.